components/utils/src/fs.rs

```diff
@@ -140,6 +141,39 @@ pub fn copy_directory(src: &Path, dest: &Path, hard_link: bool) -> Result<()> {
     Ok(())
 }
 
+pub fn copy_directory_with_ignore_globset(
+    src: &Path,
+    dest: &Path,
+    hard_link: bool,
+    gs: &GlobSet,
```

Comment:
just pass an Option<GlobSet> to `copy_directory` instead of copying the function

Reply:
done!